06_neighborhoods/Transportation/transit_trips_cost_city.R

```diff
@@ -0,0 +1,283 @@
+    county = substr(GEOID, start = 4, stop = 6),
+    tract = substr(GEOID, start = 7, stop = 12)
+  )
+
+# keep only variables of interest
+transit_trips_tracts_2015 <- transport_tracts_2015 %>%
+  select(GEOID, state, county, tract, blkgrps, population, households, transit_trips_80ami)
+transit_cost_tracts_2015 <- transport_tracts_2015 %>%
+  select(GEOID, state, county, tract, blkgrps, population, households, t_80ami)
+
+
+
+# 2019
+# bring in all the downloaded CSVs (state-level tracts) & combine them into one nation-wide file for tracts
+tracts19files <- list.files(path="C:/Users/USERNAME/Box/Lab/Projects/Gates Upward Mobility Framework/Outreach and Tools/Data/Metrics_2024_round/Transportation/2019_tract/",
+                            pattern="*.csv")
+print(tracts19files)
+tractpath19 = file.path("C:/Users/USERNAME/Box/Lab/Projects/Gates Upward Mobility Framework/Outreach and Tools/Data/Metrics_2024_round/Transportation/2019_tract",tracts19files)
+print(tractpath19)
+transport_tracts_2019 <- do.call("rbind",lapply(tractpath19,FUN=function(files){ read.csv(files)}))
+
+# create correct FIPS columns
+transport_tracts_2019 <- transport_tracts_2019 %>%
+  rename (GEOID = tract) %>%
+  mutate(
+    state = substr(GEOID, start = 2, stop = 3),
+    county = substr(GEOID, start = 4, stop = 6),
+    tract = substr(GEOID, start = 7, stop = 12)
+  )
+
+# keep only variables of interest
+transit_trips_tracts_2019 <- transport_tracts_2019 %>%
+  select(GEOID, state, county, tract, blkgrps, population, households, transit_trips_80ami)
+transit_cost_tracts_2019 <- transport_tracts_2019 %>%
+  select(GEOID, state, county, tract, blkgrps, population, households, t_80ami)
+
+
+###################################################################
+
+# 2. Crosswalk to the cities (from tracts) & aggregate data
+
+# bring in the crosswalk of interest (we want the year range to be before 2020, when tract definitions changed)
+tract_place <- read_csv("geographic-crosswalks/data/geocorr2018_tract_to_place.csv")
+# remove the decimal from the tract variable
+tract_place <- tract_place %>%
+  mutate(tract = tract*100)
+# add in leading zeroes as needed
+tract_place <- tract_place %>%
+  mutate(state = sprintf("%0.2d", as.numeric(state)),
+         tract = sprintf("%0.6d", as.integer(as.character(tract))),
+         placefp = sprintf("%0.5d", as.numeric(placefp)),
+         county = substr(county, nchar(county) - 2, nchar(county))
+  ) %>%
+  rename(place = placefp)
+
+
+# NOW, aggregate the tract-level data to the place levels for each variable for each year
+
+# For when we want to: Limit to the Census Places we want 
+# first, bring in the places crosswalk (place-populations.csv)
+places <- read_csv("geographic-crosswalks/data/place-populations.csv")
+# keep only the most relevant year
+places <- places %>%
+  filter(year == 2019)
+
+
+# TRANSPORTATION COST 2015
+transit_cost_city_2015 <- left_join(transit_cost_tracts_2015, tract_place, by=c("state", "tract"))
```

Comment:
🚨🚨🚨 I believe this merge is incorrect. 

My understanding is that you are trying to merge two datasets BY TRACT. In this case you are using a state ID and a tract ID. But tracts are unique to counties, not states, so you are potentially creating many duplicates here. The resulting dataset when I run it has 446,846 rows, whereas a proper left join using unique tracts should result in a dataframe that is 123,633 (assuming tracts are the same in both files). 

Because of this, I believe everything below that uses transit_cost_city_2015 is incorrect.

Reply:
Spot on -- new df obs count is correct, thank you for catching!

Reply:
Merge looks correct now.